tests/kafkatest/services/kafka/kafka.py

```diff
@@ -126,9 +126,14 @@ def set_version(self, version):
 
     @property
     def security_config(self):
-        return SecurityConfig(self.context, self.security_protocol, self.interbroker_security_protocol,
+        config = SecurityConfig(self.context, self.security_protocol, self.interbroker_security_protocol,
                               zk_sasl=self.zk.zk_sasl,
                               client_sasl_mechanism=self.client_sasl_mechanism, interbroker_sasl_mechanism=self.interbroker_sasl_mechanism)
+        for protocol in self.port_mappings:
+            port = self.port_mappings[protocol]
+            if port.open:
+                config.enable_security_protocol(port.name)
+        return config
```

Comment:
@ijuma Thank you for the review. The security configuration code for system test enables  client_security_protocol and interbroker_security_protocol. The files required for these enabled protocols (keystore, jaas.conf etc) are copied to the worker. The security upgrade test opens a third port for interbroker, without setting it as the interbroker_security_protocol during the first bounce. So we need to enable ssl/sasl for this as well - this change enables security protocols for all open ports.